smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/waiters/WaiterGenerator.kt

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package software.amazon.smithy.kotlin.codegen.rendering.waiters
+
+import software.amazon.smithy.kotlin.codegen.core.KotlinWriter
+import software.amazon.smithy.kotlin.codegen.core.RuntimeTypes
+import software.amazon.smithy.kotlin.codegen.core.withBlock
+import java.text.DecimalFormat
+import java.text.DecimalFormatSymbols
+
+/**
+ * Renders the top-level retry strategy for a waiter.
+ */
+private fun KotlinWriter.renderRetryStrategy(wi: WaiterInfo) {
+    setOf(
+        RuntimeTypes.Core.Retries.Impl.ExponentialBackoffWithJitterOptions,
+        RuntimeTypes.Core.Retries.Impl.ExponentialBackoffWithJitter,
+        RuntimeTypes.Core.Retries.Impl.StandardRetryStrategyOptions,
+        RuntimeTypes.Core.Retries.Impl.StandardRetryStrategy,
+        RuntimeTypes.Core.Retries.Impl.NoOpTokenBucket,
+    ).forEach(::addImport)
+
+    write("")
+    withBlock("private val #L = run {", "}", wi.retryStrategyName) {
+        withBlock("val delayOptions = ExponentialBackoffWithJitterOptions(", ")") {
+            write("initialDelayMs = #L,", wi.waiter.minDelay.msFormat)
+            write("scaleFactor = 1.5,")
+            write("jitter = 1.0,")
+            write("maxBackoffMs = #L,", wi.waiter.maxDelay.msFormat)
+        }
+        write("val delay = ExponentialBackoffWithJitter(delayOptions)")
+        write("")
+        write("val waiterOptions = StandardRetryStrategyOptions(maxTimeMs = 300_000, maxAttempts = 20)")
+        write("StandardRetryStrategy(waiterOptions, NoOpTokenBucket, delay)")
+    }
+}
+
+/**
+ * Renders the client extension methods for a waiter.
+ */
+fun KotlinWriter.renderWaiter(wi: WaiterInfo) {
+    renderRetryStrategy(wi)
```

Comment:
All of this is thread-safe...none of these types have any mutable state.

I initially had them all inline but changed it for two minor reasons:
* Since they don't change between invocations, it seemed unnecessary to instantiate them per call. This is less of a runtime optimization and more a simplification of the actual waiter method for readability sake.
* Extracting them to top-level fields makes unit testing the codegen simpler.

In the end, I don't feel super-strong about this. I can change it if you have a reason that outweighs these two.